File: 201_300/283_move_zeros/Solution.scala
object Solution {
  def moveZeroes(nums: Array[Int]): Unit = {
    var pos = 0
    var i = 0
    while (i < nums.length) {
      if (nums(i) != 0) {
        val tmp = nums(i)
        nums(i) = nums(pos)
        nums(pos) = tmp
        pos += 1
      }
      i += 1
    }
  }
}

File: 201_300/283_move_zeros/Solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int pos = 0;
        for (int i = 0; i < nums.length ; i++) {
            if (nums[i] != 0) {
                int tmp = nums[i];
                nums[i] = nums[pos];
                nums[pos++] = tmp;
            }
        }
    }
}